src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\EventListener;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Component\Stopwatch\StopwatchEvent;
+
+class KernelListener
+{
+    /**
+     * @var Stopwatch
+     */
+    protected $stopwatch;
+
+    /**
+     * @var StopwatchEvent
+     */
+    protected $stopwatchEvent;
+
+
+    /**
+     * @var RequestMetricRepository
+     */
+    protected $requestMetricRepository;
+
+    public function __construct()
+    {
+        $this->stopwatch = null;
+        $this->stopwatchEvent = null;
+    }
+
+    public function setRequestMetricRepository(RequestMetricRepository $requestMetricRepository)
+    {
+        $this->requestMetricRepository = $requestMetricRepository;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $route = $event->getRequest()->get('_route');
+
+        // Todo: check for configured routes
+        // Ignore web debug toolbar
+        if (!is_null($route) && $route !== '_wdt') {
+            $this->stopwatch = new Stopwatch();
+            $this->stopwatch->start($event->getRequest()->getUri());
+        }
+    }
+
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if (is_null($this->stopwatch)) {
+            return;
+        }
+
+        $requestUri = $event->getRequest()->getUri();
+
+        // find out if this is a profiler request, if not don't analyze it
+        try {
+            $this->stopwatchEvent = $this->stopwatch->stop($requestUri);
```

Comment:
I do check at the top of the function. Is a return value expected?

On Fri, Jun 10, 2016 at 2:55 PM, Rafał Muszyński <notifications@github.com>
wrote:

> In src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php
> <https://github.com/superdesk/web-publisher/pull/78#discussion_r66607938>:
>
> > +            $this->stopwatch->start($event->getRequest()->getUri());
> > +        }
> > +    }
> > +
> > +
> > +    public function onKernelResponse(FilterResponseEvent $event)
> > +    {
> > +        if (is_null($this->stopwatch)) {
> > +            return;
> > +        }
> > +
> > +        $requestUri = $event->getRequest()->getUri();
> > +
> > +        // find out if this is a profiler request, if not don't analyze it
> > +        try {
> > +            $this->stopwatchEvent = $this->stopwatch->stop($requestUri);
>
> then what about checking if its null and then return false? instead of
> catching the exception?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/superdesk/web-publisher/pull/78/files/e72d1ab48e970121d7c50528b7077211d4e9cc81#r66607938>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARBtFWrXrjclqy802vfXNq8Z0J0JLMf-ks5qKV6tgaJpZM4Ix7A->
> .
>